### ISSUE TITLE
Add native support for `Error.cause` in error stack traces

### ISSUE THREAD

**Issue by twschiller** [DataDog/browser-sdk#2569]
**Is your feature request related to a problem? Please describe.**

- Modern browser support nested errors with `Error.cause`: https://caniuse.com/mdn-javascript_builtins_error_cause
- The cause chain is important for error grouping and root cause diagnosis
- Rollbar added native support for `Error.cause`: https://github.com/rollbar/rollbar.js/pull/1012. We're looking to migrate from Rollbar to Datadog's log error tracking

**Describe the solution you'd like**

- The RUM and the logging SDK natively support `Error.cause`

**Describe alternatives you've considered**

- The workaround is to try fake a stack trace that Datadog supports, e.g.: https://github.com/pixiebrix/pixiebrix-extension/pull/3011/files#diff-18c5d896ecb7f366bb3e59350c17f10168a2606d32a8dcce4880f7a9a1afbfe2R259

**Comment by amortemousque** [comment]
Hello @twschiller, 
Thanks for your feedback. Which [package](https://github.com/DataDog/browser-sdk?tab=readme-ov-file#npm-packages) are you using (browser-rum or browser-log)? 
The Error.cause should be collected by default with browser-rum.

**Comment by twschiller** [comment]
Hi @amortemousque,

This is using `browser-log`. We're collecting from a Chrome Extension background page/worker (including forwarding messages from our content script). It didn't seem to make sense to use `browser-rum` in those contexts because there isn't really UI "session".

We're starting to use `browser-rum` for our extension pages (console, dev tools page, sidebar, etc.) now that the library supports local storage.

> The Error.cause should be collected

I'll give it a try with RUM and validate that it appears in the stack trace

**Comment by amortemousque** [comment]
I agree, it would make sense to also collect `Error.cause` for `browser-log`. We will keep you informed about the progress on this subject.

**Comment by BenoitZugmeyer** [comment]
This feature has been released in [v5.10.0](https://github.com/DataDog/browser-sdk/blob/main/CHANGELOG.md#v5100)
